### PR TITLE
Company addresses in search logic

### PR DESCRIPTION
**Pull request by marcofucci**
### Description of change

This uses the new company address and registered address in nested object format for the basic and company entity search.

I suggest you look at the individual commits in this PR.

### Checklist

* [x] Has a new newsfragment been created? Check [changelog/README.rst](https://github.com/uktrade/data-hub-leeloo/blob/master/changelog/README.rst) for instructions
* [ ] Have any relevant search models been updated?
* [ ] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [ ] Have any relevant select-/prefetch-related field lists in the views and search apps been updated?
* [ ] Has the admin site been updated (for new models, fields etc.)?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: datahub/search/company/test/test_elasticsearch.py [hunk 1]
```diff
@@ -561,7 +555,7 @@ def test_limited_get_search_by_entity_query():
                                     },
                                 }, {
                                     'range': {
-                                        'estimated_land_date': {
+                                        'archived': {
```
Comment (elcct) [inline datahub/search/company/test/test_elasticsearch.py]: Why this change?

Comment (reupen) [reply datahub/search/company/test/test_elasticsearch.py]: I think it's because this test is for companies, so it was a bit weird to use `estimated_land_date_after` and `estimated_land_date_before`.

File: datahub/search/contact/dict_utils.py [hunk 2]
```diff
@@ -5,11 +5,7 @@ def computed_address_field(field):
     """Gets Contact address from Company is address_same_as_company."""
     def get_field(contact):
```
Comment (reupen) [inline datahub/search/contact/dict_utils.py]: I really don't like `computed_address_field()` :cry: 

I tried to simplify it, how about something like this?

```python
def computed_address_field(field):
    """Gets Contact address from Company is address_same_as_company."""
    def get_field(contact):
        obj = contact.company if contact.address_same_as_company else contact
        value = getattr(obj, field)

        if field == 'address_country':
            return dict_utils.id_name_dict(value)

        return value

    return get_field
```

File: datahub/search/company/test/test_elasticsearch.py (hunk 1 above)
Comment (marcofucci) [reply datahub/search/company/test/test_elasticsearch.py]: yep that

File: datahub/search/contact/dict_utils.py (hunk 2 above)
Comment (marcofucci) [reply datahub/search/contact/dict_utils.py]: yes I agree, I wanted to get back as I didn't like it either. Your suggestion works well.

File: datahub/search/company/test/test_elasticsearch.py (hunk 1 above)
Comment (elcct) [reply datahub/search/company/test/test_elasticsearch.py]: Makes sense!
